python/cugraph/cugraph/dask/sampling/uniform_neighbor_sample.py

```diff
@@ -100,19 +142,47 @@ def uniform_neighbor_sample(
     with_replacement: bool, optional (default=True)
         Flag to specify if the random sampling is done with replacement
 
+    with_edge_properties: bool, optional (default=False)
+        Flag to specify whether to return edge properties (weight, edge id,
+        edge type, batch id, hop id) with the sampled edges.
+
+    batch_id_list: list (int32)
+        List of batch ids that will be returned with the sampled edges if
+        with_edge_properties is set to True.
+    
+    seed: int64, optional (default=42)
+        The random seed to use.
+        Defaults to 42.
 
     Returns
     -------
     result : dask_cudf.DataFrame
         GPU distributed data frame containing 4 dask_cudf.Series
 
-        ddf['sources']: dask_cudf.Series
-            Contains the source vertices from the sampling result
-        ddf['destinations']: dask_cudf.Series
-            Contains the destination vertices from the sampling result
-        ddf['indices']: dask_cudf.Series
-            Contains the indices from the sampling result for path
-            reconstruction
+        If with_edge_properties=True:
+            ddf['sources']: dask_cudf.Series
+                Contains the source vertices from the sampling result
+            ddf['destinations']: dask_cudf.Series
+                Contains the destination vertices from the sampling result
+            ddf['indices']: dask_cudf.Series
+                Contains the indices from the sampling result for path
+                reconstruction
+
+        If with_edge_properties=False:
+            df['sources']: dask_cudf.Series
+                Contains the source vertices from the sampling result
+            df['destinations']: dask_cudf.Series
+                Contains the destination vertices from the sampling result
+            df['edge_weight']: dask_cudf.Series
+                Contains the edge weights from the sampling result
+            df['edge_id']: dask_cudf.Series
+                Contains the edge ids from the sampling result
+            df['edge_type']: dask_cudf.Series
+                Contains the edge types from the sampling result
+            df['batch_id']: dask_cudf.Series
+                Contains the batch ids from the sampling result
+            df['hop_id']: dask_cudf.Series
+                Contains the hop ids from the sampling result
```

Comment:
Does this work in case when we dont have all the properties, like I only want want `sources `, `destinations ` , `edge_id `, `batch_id` and  `hop_id ` for DGL ?

Reply:
right now, if edge ids are provided, edge types and weights are also required

Reply:
From simpleDistributedGraph.__from_edgelist:
```
if len(edge_attr) == 1:
       input_ddf = input_ddf.rename(columns={edge_attr[0]: self.edgeWeightCol})
       value_col_names = [self.edgeWeightCol]
elif len(edge_attr) == 3:
       weight_col, id_col, type_col = edge_attr
       input_ddf = input_ddf.rename(
           columns={
               weight_col: self.edgeWeightCol,
               id_col: self.edgeIdCol,
               type_col: self.edgeTypeCol,
           }
       )

       value_col_names = [self.edgeWeightCol, self.edgeIdCol, self.edgeTypeCol]
else:
       raise ValueError("Only 1 or 3 values may be provided" "for edge_attr")
```

Reply:
Gotcha, Is there a mechanism to get `hop-ids` if I provide say only 1 column (say `edgeIdCol `) ?  The reason i am asking is the memory overhead of adding extra dummy columns will be prohibitive for large scale usecases.

Reply:
In the C++ code, either a tuple of <edge id, edge type>, or null must be provided

Reply:
But I think you can skip weights, it's just not allowed by the Python API right now.

Reply:
Also, I see that the Python API adds a weight column, even if one is not specified by the user.  I think this was done for some algorithms which require a default weight value.

Reply:
So we would need to do refactoring at both the C++ and Python API level to support passing only edge ids.

Reply:
Also, since we are now using `FeatureStore`, I thought there was less memory pressure on the sampling GPUs, giving us the flexibility to work with our current API, and eventually refactor graph creation to have less memory overhead in the future.

Reply:
Okay, I think for now, Let us stick with what we have but we will need to revisit this sooner rather than later as for large graphs memory becomes prohibitive quickly as we need 1.7x more memory, which can often mean some one requiring the double the nodes. 

The GNN use case from DGL perspective requires the following format: 
- no weights, id, no type .